packages/primitives/radio/src/Root.native.tsx

```diff
@@ -13,58 +13,56 @@ const styles = normalizeNativeStyle({
   justifyContent: 'space-between',
 })
 
+const VisibleRadio = ({ isChecked }: { isChecked: boolean}) => (
+  <View style={{
+    ...styles,
+    width: 10,
+    height: 10,
+    borderColor: '#000',
+    borderStyle: 'solid',
+    borderWidth: 1,
+    borderRadius: 50,
+    justifyContent: 'center',
+    alignItems: 'center',
+  }}
+  >
+    <View style={{
+      opacity: isChecked ? 1 : 0,
+      width: 2,
+      height: 2,
+      backgroundColor: '#000',
+      borderRadius: 50,
+      flexGrow: 0,
+    }}
+    />
+  </View>
+)
+
 export const RadioInput = component(
   startWithType<TRadioInput>(),
+  mapContext(RadioContext),
   mapDefaultProps({
     isDisabled: false,
-  }),
-  mapContext(RadioContext)
+  })
 )(({
+  isVisible,
   groupName,
-  id,
   accessibilityLabel,
-  isDisabled,
-  onChange,
-  value,
+  id,
   groupValue,
   setGroupValue,
+  value,
+  isDisabled,
+  ...props
 }) => (
   <Touchable
-    style={styles}
     testID={`${groupName}-${id}`}
     accessibilityLabel={accessibilityLabel}
-    onPress={(evt) => {
-      if (isDisabled) {
-        return false
-      }
-
-      setGroupValue(value)
-
-      if (typeof onChange === 'function') {
-        onChange(evt)
-      }
-    }}
+    onPress={() => (isDisabled ? null : setGroupValue(value))}
```

Comment:
* This generates a lot of functions during animation. Better to use `mapHandlers` from refun.
* Callback should always return `undefined`, but here it can return `null`.